src/Modules/ContentBlocks/Domain/ContentBlock/Command/DeleteContentBlockHandler.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\Command;
+
+use ForkCMS\Core\Domain\MessageHandler\CommandHandlerInterface;
+use ForkCMS\Modules\ContentBlocks\Domain\ContentBlock\ContentBlockRepository;
+use ForkCMS\Modules\Frontend\Domain\Block\Block;
+use ForkCMS\Modules\Frontend\Domain\Block\BlockRepository;
+use Exception;
+
+final class DeleteContentBlockHandler implements CommandHandlerInterface
+{
+    public function __construct(
+        private readonly ContentBlockRepository $contentBlockRepository,
+        private readonly BlockRepository $blockRepository
+    ) {
+    }
+
+    public function __invoke(DeleteContentBlock $deleteContentBlock): void
+    {
+        $versions = $this->contentBlockRepository->getVersionsForRevisionId($deleteContentBlock->id);
+
+        if (count($versions) === 0) {
+            return;
+        }
+
+        $this->contentBlockRepository->removeMultiple($versions);
+
+        $extraId = $versions[0]->getExtraId();
+        $block = $this->blockRepository->findOneBy(['id' => $extraId]);
+        $block->hide();
```

Comment:
Why `->hide`? shouldn't this delete the block?

Reply:
Delete will fail if the block is linked to a revision of a page

Reply:
Wouldn't I be more logical to disable deletion of Content blocks that are still used? So instead of showing the delete button a warning "This Content block is still in use and can not be deleted.".

Reply:
A revision of a page is not necessarily still in use

Reply:
If content blocks have revisions then delete should either delete the current and all other revisions including the block if it is not used on an active or draft page revision or archive the content block revision revision and disable the block.
I think that if we delete a content block we should properly clean it up and remove everything

Reply:
I would also disable it if it's used on a draft

Reply:
I don't understand "disable". Do you mean:
1. Delete it
2. Hide it.

If you mean `hide` I don't agree. Because the interface will look like it just can be reenabled by making it visible. But it will not work as the content is not available anymore

Reply:
Disable the delete button

Reply:
Disabling the delete button if a ContentBlock is in use on a non-archived page: https://github.com/forkcms/forkcms/pull/3555/commits/54c6cbd0d004b69c15888065472fd0c00cba3c6d
actually deleting it from the revisions still WIP

Reply:
remove contentblock: https://github.com/forkcms/forkcms/pull/3555/commits/97383ab4f9a6a347f75704a50cc198ab035564ff

Reply:
don't forget this WIP